projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.service.ts

```diff
@@ -0,0 +1,121 @@
+import { Injectable } from '@angular/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import {
+  PaymentDetails,
+  TranslationService,
+  UserPaymentService,
+} from '@spartacus/core';
+import { Card } from '../../../shared';
+import { ICON_TYPE } from '../../misc';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PaymentMethodsService {
+  loading$: Observable<
+    boolean
+  > = this.userPaymentService.getPaymentMethodsLoading();
+  editCard: string = null;
+
+  constructor(
+    protected userPaymentService: UserPaymentService,
+    protected translation: TranslationService
+  ) {
+    this.userPaymentService.loadPaymentMethods();
+  }
+
+  get(): Observable<PaymentDetails[]> {
+    return this.userPaymentService.getPaymentMethods().pipe(
+      tap((paymentDetails) => {
+        // Set first payment method to DEFAULT if none is set
+        if (
+          paymentDetails.length > 0 &&
+          !paymentDetails.find((paymentDetail) => paymentDetail.defaultPayment)
+        ) {
+          this.setDefault(paymentDetails[0]);
+        }
+      })
+    );
+  }
+
+  getCardContent({
```

Comment:
I believe destructuring in the arguments is breaking in case a null argument is passed in. Not sure if that's a potential issue.

Reply:
I also think the code isn't too elegant as the method header becomes less readable.